src/vs/workbench/services/hover/browser/hoverWidget.ts

```diff
@@ -164,16 +164,22 @@ export class HoverWidget extends Widget {
 		const targetBounds = this._target.targetElements.map(e => e.getBoundingClientRect());
 
 		// Get horizontal alignment and position
-		let targetLeft = this._target.x !== undefined ? this._target.x : Math.min(...targetBounds.map(e => e.left));
-		if (targetLeft + this._hover.containerDomNode.clientWidth >= document.documentElement.clientWidth) {
```

Comment:
FYI this was meant to protect the consumer from displaying the hover in an invalid position.

Reply:
Got it and it makes sense when `x` is being computed. If the value is coming from outside, I would assume its caller responsibility. In my case, this check is breaking the hover when sidebar is on the right. Hence removed this check when coordinates are passed.